### ISSUE TITLE
Weighted GCN layer produces some nans

### ISSUE THREAD

**Issue by Christian-Git01** [rusty1s/pytorch_geometric#795]
## 🐛 Bug

When using weighed edges as input for GCNConv-Layer, I get some nans as results which do not make any sense. The problem is connected to edge_weights. When using edge_weights=None, the problem does not occur. Changing the values of x, edge_index and edge_weights leads to different values and different positions of nans. I provide the full setup to reproduce this error below:

## To Reproduce

> import torch.nn as nn
>  import torch.nn.functional as F
> import torch_geometric
> import torch_geometric.nn as gnn
> torch.manual_seed(0)
> gcn = gnn.GCNConv(in_channels=1, out_channels=4)

> x = torch.randn(10, 1)
> edge_index = torch.randint(0, 10, [2, 22])
> edge_weight = torch.randn(22)
> y = gcn(x=x, edge_index=edge_index, edge_weight=edge_weight)
`

## Expected behavior

![image](https://user-images.githubusercontent.com/52127686/68477566-1c040c80-022e-11ea-8ff6-47f4d3fe5a8c.png)

## Environment

* OS: Linux
* Python version: Python 3.7.4
* Torch version:  1.3.0
* CUDA version: 9.0
* GCC version: 4.8.5

## Additional context

The problem is based on issue [707](https://github.com/rusty1s/pytorch_geometric/issues/707)

**Comment by rusty1s** [comment]
`GCNConv` can not handle negative edge weights.
```python
edge_weight = torch.randn(22).abs()
```
fixes the problem.

**Comment by Christian-Git01** [comment]
> `GCNConv` can not handle negative edge weights.
> 
> ```python
> edge_weight = torch.randn(22).abs()
> ```
> 
> fixes the problem.

Thank you very much! That solved my problem.

**Comment by mhorlacher** [comment]
Any reason why negative weights are not supported?

**Comment by rusty1s** [comment]
They are not supported by definition of `GCN` due to the symmetric normalization of the adjacency matrix (`deg.pow(-0.5)`).

**Comment by gsk1692** [comment]
So I stumbled upon this issue after encountering this problem. But strangely for another dataset (say A), it is working and for a second dataset B, it is not working. Is there any explanation as to why it could be working for A with negative weights?

Are there any other ways of indicating a penalty other than edge weights in PyTorch geometric?

**Comment by rusty1s** [comment]
The only reason I could think of that this works for dataset A is that you are not using any nodes with negative edge weights as readout. Might this be the case?

You may also wanna take a look at `SignedConv` and `SignedGCN` - this may be exactly what you are looking for.

**Comment by gsk1692** [comment]
Thank you for the info! I will use SignedConv if I am continuing to use the negative weights.
